docs/swagger2.yml

```diff
@@ -27,119 +27,133 @@ consumes:
+      operationId: "DeleteFn"
+      summary: "Delete a fn"
+      description: "Delete a function."
       tags:
-        - Funcs
+        - Fns
       parameters:
-        - $ref: '#/definitions/funcPathParam'
+        - $ref: '#/definitions/appIDParam'
+        - $ref: '#/definitions/fnIDParam'
       responses:
-        200:
-          description: Func successfully deleted.
+        204:
+          description: Fn successfully deleted
         404:
-          description: Func does not exist.
+          description: Fn does not exist
           schema:
             $ref: '#/definitions/Error'
         default:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
     get:
-      operationId: "GetFunc"
-      summary: "Get information for a func."
-      description: "Get information for a func."
+      operationId: "GetFn"
+      summary: "Get information for a fn"
+      description: "Get information for a Function."
       tags:
-        - Funcs
+        - Fns
       parameters:
-        - $ref: '#/definitions/funcPathParam'
+        - $ref: '#/definitions/appIDParam'
+        - $ref: '#/definitions/fnIDParam'
       responses:
         200:
-          description: Func details.
+          description: Fn metadata
           schema:
-            $ref: '#/definitions/FuncWrapper'
+            $ref: '#/definitions/FnWrapper'
         404:
-          description: Func does not exist.
+          description: Fn does not exist
           schema:
             $ref: '#/definitions/Error'
         default:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
-    patch:
+    post:
       operationId: "UpdateFunc"
-      summary: "Updates a func."
-      description: "Updates any provided fields on the func."
+      summary: "Update a func"
+      description: "Updates a Function."
       tags:
-        - Funcs
+        - Fns
       parameters:
         - $ref: '#/definitions/funcPathParam'
         - name: body
           in: body
-          description: Func object.
+          description: Fn to update
           required: true
           schema:
             $ref: '#/definitions/FuncWrapper'
       responses:
         200:
-          description: Func details.
+          description: Fn metadata
           schema:
             $ref: '#/definitions/FuncWrapper'
         400:
           description: Parameters are missing or invalid.
           schema:
             $ref: '#/definitions/Error'
-        404:
-          description: Func does not exist.
-          schema:
-            $ref: '#/definitions/Error'
         default:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
-    put:
-      operationId: "UpsertFunc"
-      summary: "Create or update a func."
-      description: "Creates a func if it does not exist, or updates the func if it does."
+
+  /apps/{app}/fns/{fn}/triggers:
```

Comment:
most list endpoints now have a filter to e.g. `/apps/app/triggers?fn=myfunctionname` to list triggers for a certain function. we can change to this if we want, we should just be consistent. less handlers / smaller api ftw to me, this has 2 spouts to list triggers per function if we add this endpoint.